src/Middleware/HttpCache.php

```diff
@@ -0,0 +1,282 @@
+
+/**
+ * HttpCache implements client-side caching by utilizing the `Last-Modified` and `ETag` HTTP headers.
+ */
+final class HttpCache implements MiddlewareInterface
+{
+    private const DEFAULT_HEADER = 'public, max-age=3600';
+    /**
+     * @internal Frozen session data
+     */
+    private ?array $sessionData;
+
+    /**
+     * @var callable a PHP callback that returns the UNIX timestamp of the last modification time.
+     * The callback's signature should be:
+     *
+     * ```php
+     * function ($request, $params)
+     * ```
+     *
+     * where `$request` is the [[ServerRequestInterface]] object that this filter is currently handling;
+     * `$params` takes the value of [[params]]. The callback should return a UNIX timestamp.
+     *
+     * @see http://tools.ietf.org/html/rfc7232#section-2.2
+     */
+    private $lastModified;
+
+    /**
+     * @var callable a PHP callback that generates the ETag seed string.
+     * The callback's signature should be:
+     *
+     * ```php
+     * function ($request, $params)
+     * ```
+     *
+     * where `$request` is the [[ServerRequestInterface]] object that this filter is currently handling;
+     * `$params` takes the value of [[params]]. The callback should return a string serving
+     * as the seed for generating an ETag.
+     */
+    private $etagSeed;
+
+    /**
+     * @var bool whether to generate weak ETags.
+     *
+     * Weak ETags should be used if the content should be considered semantically equivalent, but not byte-equal.
+     *
+     * @see http://tools.ietf.org/html/rfc7232#section-2.3
+     */
+    private bool $weakEtag = false;
+
+    /**
+     * @var mixed additional parameters that should be passed to the [[lastModified]] and [[etagSeed]] callbacks.
+     */
+    private $params;
+
+    /**
+     * @var string the value of the `Cache-Control` HTTP header. If null, the header will not be sent.
+     * @see http://tools.ietf.org/html/rfc2616#section-14.9
+     */
+    private ?string $cacheControlHeader = self::DEFAULT_HEADER;
+
+    /**
+     * @var string the name of the cache limiter to be set when [session_cache_limiter()](https://secure.php.net/manual/en/function.session-cache-limiter.php)
+     * is called. The default value is an empty string, meaning turning off automatic sending of cache headers entirely.
+     * You may set this property to be `public`, `private`, `private_no_expire`, and `nocache`.
+     * Please refer to [session_cache_limiter()](https://secure.php.net/manual/en/function.session-cache-limiter.php)
+     * for detailed explanation of these values.
+     *
+     * If this property is `null`, then `session_cache_limiter()` will not be called. As a result,
+     * PHP will send headers according to the `session.cache_limiter` PHP ini setting.
+     */
+    private ?string $sessionCacheLimiter = '';
+
+    /**
+     * @var bool a value indicating whether this filter should be enabled.
+     */
+    private bool $enabled = true;
```

Comment:
That's not needed. If you don't want it to be enabled, remove it from middleware stack.

Reply:
You are right :)